gpu_bdb/bdb_tools/cluster_startup.py

```diff
@@ -35,8 +35,24 @@ def attach_to_cluster(config, create_sql_context=False):
     scheduler_file = config.get("scheduler_file_path")
     host = config.get("cluster_host")
     port = config.get("cluster_port", "8786")
+    is_nsys_profiling_run = config.get("run_nsys_profiling", False)
+
+    if is_nsys_profiling_run:
+        from dask_cuda import LocalCUDACluster
+        cluster = LocalCUDACluster(
+            n_workers=int(os.environ.get("NUM_WORKERS", 16)),
+            device_memory_limit=os.environ.get("DEVICE_MEMORY_LIMIT", "15GB"),
+            local_directory=os.environ.get("LOCAL_DIRECTORY"),
+            rmm_pool_size=os.environ.get("POOL_SIZE", "29GB"),
+            memory_limit=os.environ.get("DEVICE_MEMORY_LIMIT", "1546828M"),
+            enable_tcp_over_ucx=True,
```

Comment:
@ayushdg I parametrized almost every input but would be good to get your inputs on the remaining ones

Reply:
Looks good to me. We might also want to set `protocol=os.environ.get("CLUSTER_MODE", "TCP")`

Reply:
Done